libbeat/processors/actions/decode_json_fields.go

```diff
@@ -0,0 +1,101 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/jsontransform"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/pkg/errors"
+)
+
+type decodeJSONFields struct {
+	Fields []string
+}
+
+var debug = logp.MakeDebug("filters")
+
+func init() {
+	processors.RegisterPlugin("decode_json_fields", configChecked(newDecodeJSONFields,
+		requireFields("fields"), allowedFields("fields", "when")))
+}
+
+func newDecodeJSONFields(c common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []string `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		logp.Warn("Error unpacking config for decode_json_fields")
+		return nil, fmt.Errorf("fail to unpack the decode_json_fields configuration: %s", err)
+	}
+
+	f := decodeJSONFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
+	var errs []string
+
+	for _, field := range f.Fields {
+		data, err := event.GetValue(field)
+		if err != nil && errors.Cause(err) != common.ErrKeyNotFound {
+			debug("Error trying to GetValue for field : %s in event : %v", field, event)
+			errs = append(errs, err.Error())
+			continue
+		}
+		text, ok := data.(string)
+		if ok {
+			var output map[string]interface{}
```

Comment:
we want to restrict ourselves to json objects only here? Using `var output interface{}` we can decode booleans, number, objects and arrays. Any valid json.